pkg/util/errutil/errhttp/writer.go

```diff
@@ -43,10 +46,44 @@ func Write(ctx context.Context, err error, w http.ResponseWriter, opts ...func(E
 
 	logError(ctx, gErr, opt)
 
+	var rsp any
 	pub := gErr.Public()
 	w.Header().Add("Content-Type", "application/json")
 	w.WriteHeader(pub.StatusCode)
-	err = json.NewEncoder(w).Encode(pub)
+	rsp = pub
+
+	// When running in k8s, this will return a v1 status
+	// Typically, k8s handlers should directly support error negotiation, however
+	// when implementing handlers directly this will maintain compatibility with client-go
+	info, ok := request.RequestInfoFrom(ctx)
+	if ok {
+		// Missing "extra" and "message.id"
```

Comment:
no clear way to add this to metav1.Status object....

🤔 we could make an object with additional fields that are ignored by client-go, but are in the response